packages/next-on-pages/src/buildApplication/processVercelFunctions/invalidFunctions.ts

```diff
@@ -309,3 +310,49 @@ async function tryToFixInvalidFuncsWithValidIndexAlternative({
 		}
 	}
 }
+
+/**
+ * Tries to fix invalid dynamic ISR functions that have valid prerendered children.
+ *
+ * The Vercel CLI might not generated a prerender config for a dynamic ISR function, depending
+ * on the Vercel CLI version. Therefore, we also check if valid prerendered routes were created
+ * for the dynamic route to determine if the function can be ignored.
+ *
+ * @param collectedFunctions Collected functions from the Vercel build output.
+ */
+async function tryToFixInvalidDynamicISRFuncs({
+	prerenderedFunctions,
+	invalidFunctions,
+	ignoredFunctions,
+}: CollectedFunctions) {
+	if (invalidFunctions.size === 0) {
+		return;
+	}
+
+	const prerenderedFunctionEntries = [...prerenderedFunctions.values()];
+
+	for (const [fullPath, fnInfo] of invalidFunctions.entries()) {
+		const fnPathWithoutRscOrFuncExt = fnInfo.relativePath.replace(
+			/(\.rsc)?\.func$/,
+			'',
+		);
+
+		const isDynamicISRFunc =
+			fnInfo.config.operationType === 'ISR' &&
+			/\/\[[\w-]+\]$/.test(fnPathWithoutRscOrFuncExt);
```

Comment:
is it because it can end with something like `.action.func`?

should be consolidate the above with:
https://github.com/cloudflare/next-on-pages/blob/f2e7bc031f4aa19561b570d04abb98669daa6b12/packages/next-on-pages/src/buildApplication/processVercelFunctions/configs.ts#L42-L44

and have a shared utility?

Reply:
it's regex for a dynamic route name

Reply:
does a dynamic route have a specific name? (is it indeed that it can't end with stuff like `.action.func`?)

what about the shared utility question? 🙂🙏

Reply:
no specific name, just standard nextjs file tree syntax `/[route]`. 
action.func shouldn't apply here since we just care about finding a dynamic route not an action

can move it to a utility sure I don't mind

Reply:
> can move it to a utility sure I don't mind

yeah, but the utility at this point makes sense only if it can be used in both places 🤔 

can we just have a utility that does something like:
```ts
return config?.operationType?.toLowerCase() === 'isr' && 
 	/\/\[[\w-]+\]$/.test(path);
```
would this work in both cases? 🤔 

if so I'd definitely say to have this utility, otherwise it's fine to keep things as they are

Reply:
Don't think so since ISR functions don't have to be dynamic

Reply:
I see... ok then never mind, thanks a bunch for the clarification 🙂